pandas/_libs/lib.pyx

```diff
@@ -2124,6 +2128,56 @@ cpdef bint is_interval_array(ndarray values):
         return False
     return True
 
+ctypedef fused maybe_convert_ret:
+    float64_t
+    uint8_t
+    uint64_t
+    int64_t
+    complex128_t
+
+cdef union NonComplexNumeric:
+    uint8_t bool_t
+    uint64_t uint_t
+    int64_t int_t
+    float64_t float_t
+
+cdef enum MaskVal:
+    bool_t = 0
+    null_t = 1
+    uint_t = 2
+    int_t = 3
+    float_t = 4
+
+
+@cython.boundscheck(False)
+@cython.wraparound(False)
+cdef void convert_union_arr_to_ndarray(NonComplexNumeric* arr,
```

Comment:
Not super familiar with this code by is there no way for us to pass a fused-type specialized memoryview or ndarray?